core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/ExcelInputFormat.java

```diff
@@ -0,0 +1,251 @@
+ */
+public class ExcelInputFormat extends TextInputFormat {
+
+  public static final String SHEET_NAME = "sheetName";
+  public static final String RE_PROCESS = "reprocess";
+  public static final String SHEET_NO = "sheetNo";
+  public static final String COLUMN_LIST = "columnList";
+  public static final String SKIP_FIRST_ROW = "skipFirstRow";
+  public static final String TERMINATE_IF_EMPTY_ROW = "terminateIfEmptyRow";
+  public static final String ROWS_LIMIT = "rowsLimit";
+  public static final String IF_ERROR_RECORD = "IfErrorRecord";
+  public static final String PROCESSED_FILES = "processedFiles";
+  public static final String FILE_PATTERN = "filePattern";
+
+  @Override
+  public RecordReader<LongWritable, Text> createRecordReader(InputSplit split, TaskAttemptContext context) {
+    return new ExcelRecordReader();
+  }
+
+  public static void setConfigurations(Job job, String filePattern, String sheetName, boolean reprocess,
+                                       int sheetNo, String columnList, boolean skipFirstRow, String terminateIfEmptyRow,
+                                       String rowLimit, String ifErrorRecord, String processedFiles) {
+
+    Configuration configuration = job.getConfiguration();
+    configuration.set(FILE_PATTERN, filePattern);
+    configuration.set(SHEET_NAME, sheetName);
+    configuration.setBoolean(RE_PROCESS, reprocess);
+    configuration.setInt(SHEET_NO, sheetNo);
+    configuration.set(COLUMN_LIST, columnList);
+    configuration.setBoolean(SKIP_FIRST_ROW, skipFirstRow);
+    configuration.set(TERMINATE_IF_EMPTY_ROW, terminateIfEmptyRow);
+
+    if (!Strings.isNullOrEmpty(rowLimit)) {
+      configuration.set(ROWS_LIMIT, rowLimit);
+    }
+
+    configuration.set(IF_ERROR_RECORD, ifErrorRecord);
+    configuration.set(PROCESSED_FILES, processedFiles);
+  }
+
+
+  /**
+   * Reads excel spread sheet, where the keys are the offset in the excel file and the text is the complete record.
+   */
+  public static class ExcelRecordReader extends RecordReader<LongWritable, Text> {
+
+    public static final String END = "END";
+    public static final String MID = "MID";
+    public static final String CELL_SEPERATION = "\t";
+    public static final String COLUMN_SEPERATION = "\r";
+
+    // Map key that represents the row index.
+    private LongWritable key;
+
+    // Map value that represents an excel row
+    private Text value;
+
+    // Represents a indvidual cell
+    private Cell cell;
+
+    // Specifies all the rows of an Excel spreadsheet - An iterator over all the rows.
+    private Iterator<Row> rows;
+
+    // InputStream handler for Excel files.
+    private FSDataInputStream fileIn;
+
+    // Path of input file.
+    private Path file;
+
+    // Specifies the row index.
+    private long rowIdx;
+
+    // Specifies last row num.
+    private int lastRowNum;
+
+    // Keeps row limits for each excel file(s)
+    private Map<String, Integer> fileRowLimitMap = new HashMap<>();
```

Comment:
Why do we need this ? Doesn't each mapper get only one file, we don't need to keep a map.